### DISCUSSION TITLE
FBA Inbound API v2024-03-20 - Deadline is December 20th -- who's still transitioning from v0?

### DISCUSSION THREAD

**Discussion by MetaMan2023** [amzn/selling-partner-api-models#4427] [Q&A]
The deadline is in 5 days... we are still working on it as there are a lot of moving pieces to program including:

* The main API (which is more complicated with more complex error handling)
* More SKU prep requirements to use the new API (more data needs the provided)
* More debug time due to slower API

Are you still working on the new API?

How far along are you?

Have you run into any roadblocks?

Do you have any backup plans if you are not complete by the 20th?

**Comment by SReynolds81** [comment]
We started on transitioning our integration in late June and still have not completed it. At this point we are somewhere around 9-10 support tickets we've had to open with Amazon due to issues with the new APIs.

I have [asked Amazon to consider delaying the deadline](https://github.com/amzn/selling-partner-api-models/issues/4048#issuecomment-2479916619), but unsurprisingly there has been no response. Someone probably has a year-end bonus riding on this deadline...

**Comment by JimMcKenzieSmith** [comment]
Yes, I'm still transitioning. I'm about 40% done.  I think around 80 hours of work remaining by my estimation... and 4 days.  I'm the only developer in the company, and I only help out on a PT contract basis, so.... ya.  Roadblocks, yes.  I get errors from the API that are undocumented.  Like "cannot be processed because the inbound plan does not support packing options."   I have no idea what that means or how to fix it.   Still thinking through backup plans.  Our system also integrates with Inventory Lab, so that's a whole new set of complexities and potential workarounds that have to be tested and thought through.

**Reply by MetaMan2023** [reply]
> cannot be processed because the inbound plan does not support packing options

There is a thread related to that error but I don't know if it provides an answer.   You can probably loop in @puppsupr 

https://github.com/amzn/selling-partner-api-models/issues/4046

How many resellers does your implementation affect?

**Comment by puppsupr** [comment]
HI @JimMcKenzieSmith,
Are you trying to follow the left hand or right hand flow per the [use-case workflow](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-use-case-guide#fulfillment-inbound-shipment-workflow) ?

**Comment by rosh3000** [comment]
In the final testing stage (aamzon-sandbox testing done).  
Crossing the tees dotting the ies

**Comment by MetaMan2023** [comment]
We are still trying to piece together the final system.  We have all of the components working but are struggling to stitch everything together by the 20th.

The hold ups for us is that, to use the left hand flow we need to have all of this information provided up front by the API to complete a shipment:

* Prep Type/Category
* Box Weight
* Box Dimensions

With our old system (v0) we did not need the new prep information (it was more limited) and the Box Weight / Dims could be manually entered later.   

We now have three different tools:

* A bulk prep tool to update 2,800 active SKUs (how many SKUs we sent in the last 6 months)
* A box weight entering tool -- for after boxes are packed to get data into our systems
* A box dimension tool -- for entering box dimensions as boxes are packed

We are working to retrain our staff to this new workflow to get this data int other system as we cannot create shipments without it and it's slowing down our transition.

Stitching all of the API calls together as well is complicated due to the error checking required and long length of time for testing.   If we want to test error conditions, it can take 3-5 minutes just to get to the point in the process where we can trigger the error to test how to handle it.

I don't feel comfortable we can be fully live by the 20th -- and I personally am traveling for the holidays and need to be hands-on on-site for trouble shooting so we are likely to stop shipping from the 21st through the first week of January.

-John

**Comment by brandondev1** [comment]
There are still so many major unresolved issues with the new version.  I can't believe they are still planning to deprecate v0.

**Comment by odd-bread** [comment]
I tried calling v0 a few minutes ago and it's still working

**Comment by puppsupr** [comment]
We have extended the final sunset date of Inbound v0 APIs from 12/20/24 to 1/21/25 for all SP-API developers. Here is the [announcement](https://developer-docs.amazon.com/sp-api/changelog/update-fulfillment-inbound-api-v0-removal-date-changed-to-january-21-2025) for the same.

**Reply by SReynolds81** [reply]
@puppsupr Thanks for letting us know.

At the end of the day, though, any sort of extension is not going to hold much value if the underlying issues with the API that made it difficult (if not impossible) for many of us to meet the original deadline are not corrected.

Does Amazon plan on releasing new updates to the new fulfillment inbound API? What is the expected release date for those changes?

**Comment by liquidone** [comment]
I am an end user, specifically a FBA only book seller. This new API is nothing short of AWFUL. Who in Amazon development though it was a good idea to make sellers wait up to an hour to get FNSKU's returned back to them?  This is ridiculous!  I don't have time to waste like this. It's like you never bothered to ask sellers about their workflow and what would make things EASIER for them. You should delay the forced migration longer then 30 days! It should be indefinite at this point. This API is NOT READY for a production environment.

**Reply by puppsupr** [reply]
We are still looking into the feasibility of this feature to be available when creating inbound plans. You need to continue to use the [JSON Listings Feeds AP](https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-use-case-guide#prerequisite-step-create-a-listing-and-convert-it-to-fba) or the Seller Central API to obtain the FNSKU.

**Reply by MetaMan2023** [reply]
> I am an end user, specifically a FBA only book seller. This new API is nothing short of AWFUL. Who in Amazon development though it was a good idea to make sellers wait up to an hour to get FNSKU's returned back to them? This is ridiculous! I don't have time to waste like this. It's like you never bothered to ask sellers about their workflow and what would make things EASIER for them. You should delay the forced migration longer then 30 days! It should be indefinite at this point. This API is NOT READY for a production environment.

We don't sell books so our operation does not need FNSKUs that quickly, however I can see how that can be an issue for a book seller (or any seller selling one-off unique items).  Out of curiosity, in the v0 system were FNSKUs returned quickly and is there a change in the new API regarding that?

**Reply by ryanowacko** [reply]
As an FBA used media seller, the labeling process significantly hinders my workflow.  Waiting an hour or more for FNSKUs and then matching them to items increases my processing time by about 5x per batch.

**Reply by puppsupr** [reply]
This is a one time process for new items to convert to FBA. Check out the steps inlined here -> https://developer-docs.amazon.com/sp-api/docs/fulfillment-inbound-api-v2024-03-20-use-case-guide#prerequisite-step-create-a-listing-and-convert-it-to-fba
